worlds/sc2wol/Locations.py

```diff
@@ -15,485 +15,710 @@ class LocationData(NamedTuple):
+                     "MissionProg",
                      lambda state: state._sc2wol_has_common_unit(multiworld, player) and
                                    state._sc2wol_has_competent_anti_air(multiworld, player) and
                                    state._sc2wol_defense_rating(multiworld, player, True) >= 3),
-        LocationData("Haven's Fall", "Haven's Fall: Northeast Colony Base", SC2WOL_LOC_ID_OFFSET + 704,
+        LocationData("Haven's Fall", "Haven's Fall: Northeast Colony Base", SC2WOL_LOC_ID_OFFSET + 704, 
+                     "Challenge",
                      lambda state: state._sc2wol_can_respond_to_colony_infestations),
-        LocationData("Haven's Fall", "Haven's Fall: East Colony Base", SC2WOL_LOC_ID_OFFSET + 705,
+        LocationData("Haven's Fall", "Haven's Fall: East Colony Base", SC2WOL_LOC_ID_OFFSET + 705, 
+                     "Challenge",
                      lambda state: state._sc2wol_can_respond_to_colony_infestations),
-        LocationData("Haven's Fall", "Haven's Fall: Middle Colony Base", SC2WOL_LOC_ID_OFFSET + 706,
+        LocationData("Haven's Fall", "Haven's Fall: Middle Colony Base", SC2WOL_LOC_ID_OFFSET + 706, 
+                     "Challenge",
                      lambda state: state._sc2wol_can_respond_to_colony_infestations),
-        LocationData("Haven's Fall", "Haven's Fall: Southeast Colony Base", SC2WOL_LOC_ID_OFFSET + 707,
+        LocationData("Haven's Fall", "Haven's Fall: Southeast Colony Base", SC2WOL_LOC_ID_OFFSET + 707, 
+                     "Challenge",
                      lambda state: state._sc2wol_can_respond_to_colony_infestations),
-        LocationData("Haven's Fall", "Haven's Fall: Southwest Colony Base", SC2WOL_LOC_ID_OFFSET + 708,
+        LocationData("Haven's Fall", "Haven's Fall: Southwest Colony Base", SC2WOL_LOC_ID_OFFSET + 708, 
+                     "Challenge",
                      lambda state: state._sc2wol_can_respond_to_colony_infestations),
-        LocationData("Smash and Grab", "Smash and Grab: Victory", SC2WOL_LOC_ID_OFFSET + 800,
+        LocationData("Smash and Grab", "Smash and Grab: Victory", SC2WOL_LOC_ID_OFFSET + 800, 
+                     "Win",
                      lambda state: state._sc2wol_has_common_unit(multiworld, player) and
                                    (logic_level > 0 and state._sc2wol_has_anti_air(multiworld, player)
                                     or state._sc2wol_has_competent_anti_air(multiworld, player))),
-        LocationData("Smash and Grab", "Smash and Grab: First Relic", SC2WOL_LOC_ID_OFFSET + 801),
-        LocationData("Smash and Grab", "Smash and Grab: Second Relic", SC2WOL_LOC_ID_OFFSET + 802),
-        LocationData("Smash and Grab", "Smash and Grab: Third Relic", SC2WOL_LOC_ID_OFFSET + 803,
+        LocationData("Smash and Grab", "Smash and Grab: First Relic", SC2WOL_LOC_ID_OFFSET + 801, 
+                     "Win"),
```

Comment:
Bonus

Reply:
It is one of the locations where a early unit can be placed. If it is removed as a location, it would break the "early unit" mechanics.

Reply:
I think this should only activate if it's the first mission containing the early unit, should be able to be turned off otherwise